init-distro-rid.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+# initNonPortableDistroRid
+#
+# Input:
+#   isCrossBuild? (nullable vararg:int)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# initNonPortableDistroRid will attempt to initialize a non portable rid. These
+# rids are specific to distros need to build the product/package and consume
+# them on the same platform.
+#
+# Below is the list of current non-portable platforms.
+#
+# Builds from the following *must* be non-portable:
+#
+#   |    OS     |           Expected RID            |
+#   -------------------------------------------------
+#   |   rhel6   |           rhel.6-x64              |
+#   |  alpine*  |        linux-musl-(arch)          |
```

Comment:
We should change alpine to the same plan as the other linux distros. In other words, the non portable rid should be alpine.(version)-(arch) and the corresponding portable rid linux-musl-(arch). Also, the default build flavor for Alpine should be set to portable. We already did the build flavor change in corefx, so this would complete it.

Reply:
@janvorli @echesakovMSFT and I have a bit of confusion around this terminology. To clarify a "portable" alpine build is defined to be portable between alpine versions. This is different from our portable definition. Our normal portable definition is a portable build is portable between any portable linux os or version.

This being said, why do we want to support version specific portability in alpine only? In addition, what is the benefit to also allowing non-version specific non-portable builds as well?

Reply:
Addressed

Reply:
> why do we want to support version specific portability in alpine only

I think this is because there are other non-Alpine distros that ship with musl-libc. One such distro is VoidLinux (with both glibc and musl-libc editions), where we can use .NET Core 3.0 preview today: [docker](https://hub.docker.com/r/voidlinux/voidlinux-musl) or [wsl](https://github.com/am11/VoidMuslWSL). e.g.

```sh
export CLR_OPENSSL_VERSION_OVERRIDE=46

xbps-install -S
xbps-install -y curl icu gettext

DotnetDir=~/dotnet-3.0-preview
mkdir $DotnetDir
curl https://download.visualstudio.microsoft.com/download/pr/3e28fec3-fd16-4fb3-8eaf-bb7351ab210c/50b758004c5c307fa048b7ef41fa47d2/dotnet-sdk-3.0.100-preview-010184-linux-musl-x64.tar.gz | tar -xz -C $DotnetDir

$DotnetDir/dotnet new console -n test-hello
$DotnetDir/dotnet run -p test-hello
```